worlds/kdl3/Options.py

```diff
@@ -0,0 +1,387 @@
+    """
+    The color of Kirby, from a list of presets.
+    """
+    display_name = "Kirby Flavor"
+    option_default = 0
+    option_bubblegum = 1
+    option_cherry = 2
+    option_blueberry = 3
+    option_lemon = 4
+    option_kiwi = 5
+    option_grape = 6
+    option_chocolate = 7
+    option_marshmallow = 8
+    option_licorice = 9
+    option_watermelon = 10
+    option_orange = 11
+    option_lime = 12
+    option_lavender = 13
+    option_custom = 14
+    default = 0
+
+    @classmethod
+    def from_text(cls, text: str) -> Choice:
+        if text == "random":
+            choice_list = list(cls.name_lookup)
+            choice_list.remove(14)
+            return cls(random.choice(choice_list))
+        return super().from_text(text)
+
+
+class KirbyFlavor(OptionDict):
+    """
+    A custom color for Kirby. To use a custom color, set the preset to Custom and then define a dict of keys from "1" to
+    "15", with their values being an HTML hex color.
+    """
+    default = {}
+
+
+class GooeyFlavorPreset(Choice):
+    """
+    The color of Gooey, from a list of presets.
+    """
+    display_name = "Gooey Flavor"
+    option_default = 0
+    option_bubblegum = 1
+    option_cherry = 2
+    option_blueberry = 3
+    option_lemon = 4
+    option_kiwi = 5
+    option_grape = 6
+    option_chocolate = 7
+    option_marshmallow = 8
+    option_licorice = 9
+    option_watermelon = 10
+    option_orange = 11
+    option_lime = 12
+    option_lavender = 13
+    option_custom = 14
+    default = 0
+
+    @classmethod
+    def from_text(cls, text: str) -> Choice:
+        text = text.lower()
+        if text == "random":
+            choice_list = list(cls.name_lookup)
+            choice_list.remove(14)
+            return cls(random.choice(choice_list))
+        return super().from_text(text)
+
+
+class GooeyFlavor(OptionDict):
+    """
+    A custom color for Gooey. To use a custom color, set the preset to Custom and then define a dict of keys from "1" to
+    "15", with their values being an HTML hex color.
+    """
+    default = {}
+
+
+class MusicShuffle(Choice):
+    """
+    None: default music will play
+    Shuffled: music will be shuffled amongst each other
+    Full: random music will play in each room
+    Note that certain songs will not be chosen in shuffled or full
+    """
+    display_name = "Music Randomization"
+    option_none = 0
+    option_shuffled = 1
+    option_full = 2
+    default = 0
+
+
+class VirtualConsoleChanges(Choice):
+    """
+    Adds the ability to enable 2 of the Virtual Console changes.
+    Flash Reduction: reduces the flashing during the Zero battle.
+    Color Changes: changes the color of the background within the Zero Boss Butch rematch.
+    """
+    display_name = "Virtual Console Changes"
+    option_none = 0
+    option_flash_reduction = 1
+    option_color_changes = 2
+    option_both = 3
+    default = 1
+
+
+kdl3_options: typing.Dict[str, type(Option)] = {
```

Comment:
https://docs.python.org/3/library/typing.html#the-type-of-class-objects
Type annotations should use square brackets `type[Option]` and not call the `type` function.
But `type[Option]` isn't supported in Python 3.8, so in Python 3.8, it should be `typing.Type[Option]`.

Or, since we have a name for this metaclass, it could be `AssembleOptions`.

But `typing.Type[Option]` might be better understood by readers.

Reply:
Actually, since `Option` is `Generic`, a strict type checker would like to see `typing.Type[Option[typing.Any]]`
(or `AssembleOptions`)